src/pip/_internal/network/cache.py

```diff
@@ -77,6 +77,21 @@ def _write(self, path: str, data: bytes) -> None:
             with adjacent_tmp_file(path) as f:
                 f.write(data)
 
+                # Change permissions only if there is no risk of following a symlink
+                if (
+                    os.chmod in os.supports_fd
+                    or os.chmod in os.supports_follow_symlinks
```

Comment:
I think you may have misinterpreted the documentation, which to be fair does have some vagueness to what "it" is referring to. In the paragraph of the documentation in question,
>You can check whether or not path can be specified as a file descriptor for ***a particular function*** on your platform using os.supports_fd. If this functionality is unavailable, using ***it*** will raise a NotImplementedError.

(emphasis mine), I believe the "it" is referring to using the particular function with the given functionality when it is not available, and not using the `os.supports_*` sets to check if the functionality is available for that function. The whole point in checking them is to avoid throwing those exceptions. Furthermore, the actual part of the documention for the various `os.supports_*` sets make no mention of the possibility that using the set could throw such an error (for example [os.supports_fd documentation](https://docs.python.org/3/library/os.html#os.supports_fd)).

Reply:
Thanks, I agree, on a careful reread I misinterpreted what the documentation meant by "it".